src/main/java/com/pingcap/tikv/codec/TableCodec.java

```diff
@@ -89,6 +145,58 @@ public static long decodeRowKey(ByteString rowKey) {
     return IntegerType.readLong(cdi);
   }
 
+  public static class DecodeResult {
+    public long handle;
+    public enum Status {
+      MIN,
+      MAX,
+      EQUAL,
+      LESS,
+      GREATER,
+      UNKNOWN_INF
+    }
+    public Status status;
+  }
+
+  public static void tryDecodeRowKey(long tableId, byte[] rowKey, DecodeResult outResult) {
```

Comment:
what will happen if rowKey is invalid for decoding? Will it be always decodable?

Reply:
Yeah. We assume `rowKey` is always decidable. It is rarely happened in reality because all `rowKey` comes from `Spark` side. `rowKey` is undecidable only if `tikv` encodes it wrongly. However, it is better to have an exception.

Reply:
yes. since last part we try decode it as long which is always success operation. If length is not enough, we padding it to be full length of long. And return value will be marked as less/greater if remaining key length is not enough or larger than a long type for further comparison needs.

Reply:
Understood, thx for detailed explanation.